source/main.cpp

```diff
@@ -204,34 +235,20 @@ Connection: close
         printf("Gateway: %s\r\n", a.get_ip_address() ? a.get_ip_address() : "None");
     }
 
-    static const char *get_security_string(nsapi_security_t sec)
-    {
-        switch (sec) {
-            case NSAPI_SECURITY_NONE:
-                return "None";
-            case NSAPI_SECURITY_WEP:
-                return "WEP";
-            case NSAPI_SECURITY_WPA:
-                return "WPA";
-            case NSAPI_SECURITY_WPA2:
-                return "WPA2";
-            case NSAPI_SECURITY_WPA_WPA2:
-                return "WPA/WPA2";
-            case NSAPI_SECURITY_UNKNOWN:
-            default:
-                return "Unknown";
-        }
-    }
-
 private:
     NetworkInterface *_net;
-    TCPSocket socket;
+
+#if MBED_CONF_APP_USE_TLS_SOCKET
+    TLSSocket _socket;
+#else
+    TCPSocket _socket;
+#endif // MBED_CONF_APP_USE_TLS_SOCKET
 };
 
 int main() {
-    SocketDemo example;
-
     printf("\r\nStarting socket demo\r\n\r\n");
 
-    example.run();
+    SocketDemo *example = new SocketDemo;
```

Comment:
Why using dynamic memory allocation when on-stack allocation would work just as well ?

Reply:
running out of stack memory